### PR TITLE
use stable PHP 8 releases in the CI

### PR DESCRIPTION
**Pull request by xabbuh**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,17 +28,16 @@ matrix:
     # development dependencies
     - php: 7.4
       env: DEPENDENCIES='dev'
-    - php: nightly
-      env: COMPOSER_FLAGS="--ignore-platform-reqs" SYMFONY_DEPRECATIONS_HELPER="max[self]=0"
+    - php: 8.0
```
Comment (GuilhemN) [inline .travis.yml]: It seems `8.0snapshot` is in use in EasyAdminBundle, I suppose we could do the same

**Comment by GuilhemN** [comment]
I pushed some commits to your branch, I hope you don't mind.

I upgraded the code to phpunit 9 as it's the only version supporting php 8. The ignore requirements is back due to JMS\serializer.
There are some failures remaining, I'm not sure yet where do they come from, they may be related to some changes in php 8.

**Comment by GuilhemN** [comment]
Wow the battle was tough... But I won against phpunit and php 8 :D 

This is ready for me, I can also do the merge in 3.x if you want.

File: Negotiation/FormatNegotiator.php [hunk 2]
```diff
@@ -40,7 +40,7 @@ public function add(RequestMatcherInterface $requestMatcher, array $options = []
         $this->map[] = [$requestMatcher, $options];
     }
 
-    public function getBest($header, array $priorities = [])
+    public function getBest($header, array $priorities = [], $strict = false)
```
Comment (kevinpapst) [inline Negotiation/FormatNegotiator.php:43]: seems to be an unnecessary change

Comment (GuilhemN) [reply Negotiation/FormatNegotiator.php:43]: it is a necessary change as `willdurand/negotiation` changed the signature of this method in v3 and we have to update it too to comply with the change, otherwise php raises an error.

Comment (kevinpapst) [reply Negotiation/FormatNegotiator.php:43]: Oh, ups. I checked the complete codebase ... but missed to check the base class. Thanks for the explanation & sorry for stealing your time

Comment (GuilhemN) [reply Negotiation/FormatNegotiator.php:43]: No problem, it's very nice of you to review these changes :)

Comment (kevinpapst) [reply Negotiation/FormatNegotiator.php:43]: The least I can do if relying on this bundle^^ and btw: my projects api test suite is green with this branch (around 100 integration tests). 
Looking forward for a PHP 8 enabled release! 😁

**Comment by DenchikBY** [comment]
Hi, do you plan rework annotations in this PR to support php 8 format?

**Comment by xabbuh** [comment]
If you are talking about support for PHP 8 attributes, that's not in the scope of this PR. But if you are interested in giving it a try, please feel free to open a separate pull request.

**Comment by kevinpapst** [comment]
Can we help to push this PR and the upcoming release that will include it?

**Comment by GuilhemN** [comment]
Oops sorry I just realized this PR was also targetting 2.x and I could have completed it instead of pushing to https://github.com/FriendsOfSymfony/FOSRestBundle/pull/2280... I guess I wasn't well awaken when I did that :x

Anyway it's merged now so too late I suppose... I'm closing this PR in favor of https://github.com/FriendsOfSymfony/FOSRestBundle/pull/2280 :)

File: Negotiation/FormatNegotiator.php (hunk 2 above)
Comment (willdurand) [reply Negotiation/FormatNegotiator.php:43]: ah yeah sorry for the breaking change here.. hopefully it wasn't too complicated to migrate

Comment (GuilhemN) [reply Negotiation/FormatNegotiator.php:43]: No worry, it was manageable :) Thank you for your work on this library!
